src/sys/stm_tests.ml

```diff
@@ -0,0 +1,242 @@
+      else touch_model fs path new_file_name perm
+
+  let init_sut () =
+    match Sys.os_type with
+    (* | "Unix" -> ignore (Sys.command ("rm -rf " ^ (static_path / "sandbox_root") ^ " && mkdir " ^ (static_path / "sandbox_root"))) *)
+    | "Unix" -> ignore (Sys.command ("mkdir " ^ (static_path / "sandbox_root")))
+    (* | "Win32" ->
+      ignore (Sys.command (
+      "powershell -Command \"Remove-Item -Path " ^ (static_path / "sandbox_root") ^ " -Recurse -Force -ErrorAction Ignore \"
+       & mkdir " ^ (static_path / "sandbox_root"))) *)
+    | "Win32" -> ignore (Sys.command ("mkdir " ^ (static_path / "sandbox_root")))
+    | v -> failwith ("Sys tests not working with " ^ v)
+
+  let cleanup _ =
+    match Sys.os_type with
+    | "Unix" -> ignore (Sys.command ("rm -rf " ^ (static_path / "sandbox_root")))
+    | "Win32" -> ignore (Sys.command ("powershell -Command \"Remove-Item '" ^ (static_path / "sandbox_root") ^ "' -Recurse -Force\""))
+    | v -> failwith ("Sys tests not working with " ^ v)
+
+  let precond _c _s = true
+
+  let p path =  (List.fold_left (/) (static_path / "sandbox_root") path)
+
+  let run c _file_name =
+    match c with
+    | File_exists path -> Res (bool, Sys.file_exists (p path))
+    | Mkdir (path, new_dir_name, perm) ->
+      Res (result unit exn, protect (Sys.mkdir ((p path) / new_dir_name)) perm)
+    | Rmdir (path, delete_dir_name) ->
+      Res (result unit exn, protect (Sys.rmdir) ((p path) / delete_dir_name))
+    | Readdir path ->
+      Res (result (array string) exn, protect (Sys.readdir) (p path))
+    | Touch (path, new_file_name, _perm) ->
+        (match Sys.os_type with
+        | "Unix" -> Res (unit, ignore(Sys.command ("touch " ^ (p path) / new_file_name ^ " 2>/dev/null")))
+        | "Win32" -> Res (unit, ignore(Sys.command ("type nul >> \"" ^ (p path / new_file_name) ^ "\"")))
+        | v -> failwith ("Sys tests not working with " ^ v))
+
+  let fs_is_a_dir fs = match fs with | Directory _ -> true | File _ -> false
+
+  let path_is_a_dir fs path =
+    match find_opt_model fs path with
+    | None -> false
+    | Some target_fs -> fs_is_a_dir target_fs
+
+  let postcond c (fs: filesys) res =
+    match c, res with
+    | File_exists path, Res ((Bool,_),b) -> b = mem_model fs path
+    | Mkdir (path, new_dir_name, _perm), Res ((Result (Unit,Exn),_), res) ->
+      let complete_path = (path @ [new_dir_name]) in
+      (match res with
+       | Error err ->
+         (match err with
+           | Sys_error s ->
+             (s = (p complete_path) ^ ": Permission denied") ||
+             (s = (p complete_path) ^ ": File exists" && mem_model fs complete_path) ||
+             (s = (p complete_path) ^ ": No such file or directory" && not (mem_model fs path)) ||
+             (s = (p complete_path) ^ ": Not a directory" && not (path_is_a_dir fs complete_path))
+           | _ -> false)
+        | Ok () -> not (mem_model fs complete_path) && mem_model fs path && path_is_a_dir fs path)
```

Comment:
Here I would suggest putting the `not (mem_model ...)` case last - as it reads more naturally